uarray/_uarray_dispatch.cxx

```diff
@@ -353,6 +353,80 @@ struct SetBackendContext
     }
 };
 
+struct Dispatchable
+{
+  PyObject_HEAD
+
+  py_ref value;
+  py_ref type;
+  bool coercible;
+
+  static void dealloc(Dispatchable * self)
+  {
+    self->~Dispatchable();
+    Py_TYPE(self)->tp_free(self);
+  }
+
+  static PyObject * get_value(Dispatchable * self, void *)
+  {
+    return py_ref::ref(self->value).get();
```

Comment:
This creates a temporary `py_ref` that is almost immediately destroyed, so the reference count doesn't actually change. You could either: use `release` instead of `get` to indicate ownership transfer; or just explicily `INCREF`
```cpp
Py_INCREF(self->value);
return self->value;
```